crates/uv-resolver/src/error.rs

```diff
@@ -94,20 +97,20 @@ pub enum ResolveError {
     ParsedUrl(#[from] uv_pypi_types::ParsedUrlError),
 
     #[error("Failed to download `{0}`")]
-    Download(Box<BuiltDist>, #[source] uv_distribution::Error),
+    Download(Box<BuiltDist>, #[source] Arc<uv_distribution::Error>),
```

Comment:
I don't think I can avoid this `Arc`... I need to be able to clone this, because it's stored in the `OnceMap`, but then needs to be cloned to be returned as a `Result` in the resolver. And `uv_distribution::Error` doesn't implement `Clone`.

Reply:
`Arc` inside an error is totally natural: https://github.com/BurntSushi/jiff/blob/16376c7175bca14a8ccb37a178bde92598af1a5f/src/error.rs#L47-L57

The main alternative here, and one I would suggest if possible, is to make `uv_distribution::Error` embed an `Arc` internally so that it can implement `Clone` itself. Then you don't need to spread `Arc`s out everywhere else.